### PR TITLE
Improve radial menu

### LINKED ISSUE #57: Improve radial menu
- Make semi transparent when menu is not being interacted with via ray or analog
- Reduce overall size
- Center around the bottom/back of the radial menu region; buttons shouldn't populate from the top down along the left side, but from the bottom center outward (less important, but would present better)
- Make semitransparent when objects are direct selected
- If user interacts with menu while direct selecting, radial menu regains typical transparency

### PR DESCRIPTION
**Pull request by dunity**
-Make semi transparent when menu is not being interacted with via ray or analog
-Reduce overall size
-Center around the bottom/back of the radial menu region; buttons shouldn't populate from the top down along the left side, but from the bottom center outward (less important, but would present better)
-Make semitransparent when objects are direct selected
-If user interacts with menu while direct selecting, radial menu regains typical transparency

Resolves #57

### CODE REVIEW

**Comment by dunity** [comment]
I'll perform a merge of current development(tooltips now merged in) into this branch, and fix a recently identified bug with workspaces and the new hiding functionality added in this PR.

**Comment by dunity** [comment]
-The quick hide/unhide bug has been fixed (thanks for identifying it @mtschoen )
-Refactored 2 existing functions into single Visibility property in RadialMenuSlot
-Current development branch merged into this branch; New Tooltip changes merged into the existing RadialMenuSlot prefab changes.

File: Menus/RadialMenu/Scripts/RadialMenuSlot.cs [hunk 1]
```diff
@@ -11,12 +11,15 @@ namespace UnityEngine.Experimental.EditorVR.Menus
...
 	public class RadialMenuSlot : MonoBehaviour, ISetTooltipVisibility, ITooltip, ITooltipPlacement, IRayEnterHandler, IRayExitHandler
 	{
-		static readonly Vector3 kHiddenLocalScale = new Vector3(1f, 0f, 1f);
-		const float m_IconHighlightedLocalYOffset = 0.006f;
-		const string kMaterialAlphaProperty = "_Alpha";
-		const string kMaterialExpandProperty = "_Expand";
-		const string kMaterialColorTopProperty = "_ColorTop";
-		const string kMaterialColorBottomProperty = "_ColorBottom";
+		static Color s_FrameOpaqueColor;
+
```
Comment (amirebrahimi) [inline Menus/RadialMenu/Scripts/RadialMenuSlot.cs]: Remove extra line

File: Menus/RadialMenu/Scripts/RadialMenuSlot.cs [hunk 2]
```diff
@@ -225,21 +277,25 @@ IEnumerator AnimateShow()
...
-			m_CanvasGroup.interactable = true;
 			transform.localScale = Vector3.one;
-
-			CorrectIconRotation();
+			ShowCleanup();
 
 			m_VisibilityCoroutine = null;
 		}
 
+		void ShowCleanup()
```
Comment (amirebrahimi) [inline Menus/RadialMenu/Scripts/RadialMenuSlot.cs]: What does "Cleanup" mean in this context?

File: Menus/RadialMenu/Scripts/RadialMenuSlot.cs [hunk 3]
```diff
@@ -349,44 +414,108 @@ void SetIconPressed()
...
+		{
+			if (m_InsetRevealCoroutine != null)
+			{
+				// In case semiTransparency is triggered immedlately upon showing the radial menu
+				this.StopCoroutine(ref m_InsetRevealCoroutine);
+				m_CanvasGroup.alpha = 1f;
+				ShowCleanup();
+			}
+
+			const float k_FasterMotionMultiplier = 2f;
```
Comment (amirebrahimi) [inline Menus/RadialMenu/Scripts/RadialMenuSlot.cs]: Don't use k_ for local variables. You're free to use k (no underscore or simply camel case naming with no prefix. Please fix up any other areas where you have done this.

File: Menus/RadialMenu/Scripts/RadialMenuUI.cs [hunk 4]
```diff
@@ -204,13 +219,15 @@ public void Setup()
 
 		void SetupRadialSlotPositions()
 		{
-			const float kRotationSpacing = 360f / kSlotCount;
-			for (int i = 0; i < kSlotCount; ++i)
+			const float k_RotationSpacing = 360f / k_SlotCount;
```
Comment (amirebrahimi) [inline Menus/RadialMenu/Scripts/RadialMenuUI.cs]: Rename

File: Menus/RadialMenu/RadialMenu.cs [hunk 5]
```diff
@@ -70,6 +71,7 @@ public bool visible
 		public Transform menuOrigin { get; set; }
 
 		public GameObject menuContent { get { return m_RadialMenuUI.gameObject; } }
+		public Func<Dictionary<Transform, DirectSelectionData>> getDirectSelection { private get; set; }
```
Comment (amirebrahimi) [inline Menus/RadialMenu/RadialMenu.cs]: Remove

File: Menus/RadialMenu/RadialMenu.cs [hunk 6]
```diff
@@ -78,6 +80,7 @@ void Start()
 			m_RadialMenuUI.actions = menuActions;
 			m_RadialMenuUI.Setup();
 			m_RadialMenuUI.visible = m_Visible;
+			m_RadialMenuUI.getDirectSelection = getDirectSelection;
```
Comment (amirebrahimi) [inline Menus/RadialMenu/RadialMenu.cs]: Remove

File: Menus/RadialMenu/Scripts/RadialMenuUI.cs [hunk 7]
```diff
@@ -30,6 +27,9 @@ public class RadialMenuUI : MonoBehaviour, IConnectInterfaces
 		List<RadialMenuSlot> m_RadialMenuSlots;
 		Coroutine m_VisibilityCoroutine;
 		RadialMenuSlot m_HighlightedButton;
+		float m_PhaseOffset; // Correcting the coordinates, based on actions count, so that the menu is centered at the bottom
+
+		public Func<Dictionary<Transform, DirectSelectionData>> getDirectSelection { private get; set; }
```
Comment (amirebrahimi) [inline Menus/RadialMenu/Scripts/RadialMenuUI.cs]: Remove

**Pull request merged**